mpas_analysis/ocean/time_series_antarctic_melt.py

```diff
@@ -0,0 +1,238 @@
+                          '{}.'.format(streamName, self.startDate,
+                                       self.endDate))
+
+        return  # }}}
+
+    def run(self):  # {{{
+        """
+        Performs analysis of the time-series output of Antarctic sub-ice-shelf
+        melt rates.
+
+        Authors
+        -------
+        Xylar Asay-Davis
+        """
+
+        print "\nPlotting Antarctic melt rate time series..."
+
+        print '  Load melt rate data...'
+
+        config = self.config
+        calendar = self.calendar
+
+        print '\n  Reading files:\n' \
+              '    {} through\n    {}'.format(
+                  os.path.basename(self.inputFiles[0]),
+                  os.path.basename(self.inputFiles[-1]))
+
+        # Load data:
+        variableList = ['timeMonthly_avg_landIceFreshwaterFlux']
+        timeVariableName = ['xtime_startMonthly', 'xtime_endMonthly']
+        ds = open_multifile_dataset(fileNames=self.inputFiles,
+                                    calendar=calendar,
+                                    config=config,
+                                    timeVariableName=timeVariableName,
+                                    variableList=variableList,
+                                    startDate=self.startDate,
+                                    endDate=self.endDate)
+
+        freshwaterFlux = ds.timeMonthly_avg_landIceFreshwaterFlux
+
+        movingAverageMonths = config.getint('timeSeriesAntarcticMelt',
+                                            'movingAverageMonths')
+
+        iceShelvesToPlot = config.getExpression('timeSeriesAntarcticMelt',
+                                                'iceShelvesToPlot')
+
+        dsRestart = xarray.open_dataset(self.restartFileName)
+        areaCell = dsRestart.areaCell
+
+        dsRegionMask = xarray.open_dataset(self.regionMaskFileName)
+        regionNames = list(dsRegionMask.regionNames.values)
+        nRegions = dsRegionMask.dims['nRegions']
+
+        if 'all' in iceShelvesToPlot:
+            iceShelvesToPlot = regionNames
+            regionIndices = [iRegion for iRegion in range(nRegions)]
+        else:
+            regionIndices = []
+            for regionName in iceShelvesToPlot:
+                if regionName not in regionNames:
+                    raise ValueError('Unknown ice shelf name {}'.format(
+                            regionName))
+
+                iRegion = regionNames.index(regionName)
+                regionIndices.append(iRegion)
+
+        # select only those regions we want to plot
+        dsRegionMask = dsRegionMask.isel(nRegions=regionIndices)
+        cellMasks = dsRegionMask.regionCellMasks
+        nRegions = dsRegionMask.dims['nRegions']
+
+        # convert from kg/s to kg/yr
+        totalMeltFlux = constants.sec_per_year * \
+            (cellMasks*areaCell*freshwaterFlux).sum(dim='nCells')
+
+        totalArea = (cellMasks*areaCell).sum(dim='nCells')
```

Comment:
@xylar -- I did some spot checking here of places where either melt flux or average melt rate from our simulations roughly agrees with Rignot et al. One or the other is close in a few places, but never both, and this appears to be because the shelf areas based on the cellMasks are too large in all cases I've checked (by 30-50%). Overall, the total shelf area estimate for all of Antarctica is ~2.7x too large relative to Rignot et al. I can imagine how this could be if we are always treating a grid cell as floating if any part of it is partially floating. Is that the case? This problem will get better at higher resolution, but it may make comparing modeled and obs. melt rates difficult for our lower-res. simulations. It's possible there is some other bug here as well, but I thought I'd get your thoughts before digging too deep on this. Thanks.

Reply:
Sorry, @stephenprice, I am missing a factor here.  `freshwaterFlux` should be multiplied by `landIceFraction`, since many cells are only partly covered by land ice.  I'll make a note of this here and fix it momentarily.

Reply:
I noticed that the numbers seemed weird but I didn't realize how bad the problem was or what the cause was until your comment.